packages/odf/components/object-service-nav-item/object-service.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { NooBaaObjectBucketClaimModel } from '@odf/core/models';
+import {
+  HorizontalNavProps as ObjectServiceTabProps,
+  convertHorizontalNavTabToNav as convertObjectServiceTabToNav,
+  useSortPages,
+} from '@odf/core/utils';
+import {
+  HorizontalNavTab,
+  isHorizontalNavTab,
+} from '@odf/odf-plugin-sdk/extensions';
+import { StatusBox } from '@odf/shared/generic/status-box';
+import PageHeading from '@odf/shared/heading/page-heading';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { referenceForModel } from '@odf/shared/utils';
+import {
+  HorizontalNav,
+  useResolvedExtensions,
+  NamespaceBar,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Extension,
+  ExtensionTypeGuard,
+} from '@openshift-console/dynamic-plugin-sdk/lib/types';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+import { RouteComponentProps, match as Match } from 'react-router';
+import { useHistory } from 'react-router';
+import { useLocation } from 'react-router-dom';
+import { compose } from 'redux';
+
+const OBJECT_SERVICE_CONTEXT = 'odf-object-service';
+const NAMESPACE_BAR_PATHS = [referenceForModel(NooBaaObjectBucketClaimModel)];
+
+const ObjectServicePage: React.FC = () => {
+  const { t } = useCustomTranslation();
+  const title = t('Object Service');
+  const [pages, setPages] = React.useState<ObjectServiceTabProps[]>([]);
+
+  const isObjectServiceTab = React.useMemo(
+    () =>
+      compose(isHorizontalNavTab, (e: Extension) => {
+        if (e.properties.contextId === OBJECT_SERVICE_CONTEXT) return e;
+      }),
+    []
+  );
+
+  const [extensions, isLoaded, error] = useResolvedExtensions<HorizontalNavTab>(
+    isObjectServiceTab as ExtensionTypeGuard<HorizontalNavTab>
+  );
+
+  // When the list of matching extensions changes, the resolution is restarted.
+  // The hook will continue to return the previous result until the resolution completes.
+  // Which means that value of "haveExtensionsResolved" can change later when new extensions are being loaded.
+  const haveExtensionsResolved =
+    isLoaded && _.isEmpty(error) && !_.isEmpty(extensions);
+  const haveAlreadyResolvedOnce = React.useRef<Boolean>(false);
+  if (!haveAlreadyResolvedOnce.current && haveExtensionsResolved) {
+    haveAlreadyResolvedOnce.current = true;
+  }
+  useSortPages({ extensions, haveExtensionsResolved, pages, setPages });
```

Comment:
```suggestion
  const pages = useSortPages({ extensions, haveExtensionsResolved});
```
The root state of all these pages are the extensions which is stored in a state somewhere. If we are to use a hook then we can potentially avoid keeping an intermeddiate state to store pages. WDYT?

Reply:
make sense !